pypsa/optimization/constraints.py

```diff
@@ -425,9 +429,9 @@ def p_previous(idx):
         m.add_constraints(lhs, "<=", rhs, f"{c}-com-{attr}-ramp_limit_up", mask=mask)
 
     # com down
-    if not assets.ramp_limit_down.isnull().all():
+    if not assets[["ramp_limit_down", "ramp_limit_shut_down"]].isnull().all().all():
         limit_shut = assets.eval("ramp_limit_shut_down * p_nom").to_xarray()
-        limit_down = assets.eval("ramp_limit_down * p_nom").to_xarray()
+        limit_down = assets.eval("ramp_limit_down * p_nom").fillna(1.0).to_xarray()
```

Comment:
I am wondering whether the fill with 1 makes sense here, since this is in units of MW.

Reply:
no it has to be filled with then "p_nom". Thanks for pointing it out. I already corrected it but haven't pushed it yet. I am currently having troubles since the problem seems to be unbounded with the new formulation of the constraint